### PR TITLE
Add initial capability for verb-noun commands.

### PR DESCRIPTION
**Pull request by geomacy**
This adds the capability of the CLI doing verb-noun style commands, using
codegansta's SubCommand capability.  By verb-noun we mean commands like
"list application" rather than "application list".

This commit adds code to support verb-noun, and does a proof-of-concept for
it by re-implementing the "applications" command as "list application". For
the moment it does not remove "applications".

For example:

$ bk create ~/myweb.yaml
response Status: 201 Created
...
$ bk applications
Id         Name             Status    Location
aoCeWY0f   My Web Cluster   RUNNING   LXEMPWA4
$
$ bk list application
Id         Name             Status    Location
aoCeWY0f   My Web Cluster   RUNNING   LXEMPWA4

### CODE REVIEW

File: commands/applications.go [hunk 1]
```diff
@@ -21,9 +21,9 @@ func NewApplications(network *net.Network) (cmd *Applications) {
 
 func (cmd *Applications) Metadata() command_metadata.CommandMetadata {
 	return command_metadata.CommandMetadata{
-		Name:        "applications",
+		Name:        "application",
 		Description: "Show the status and location of running applications",
-		Usage:       "BROOKLYN_NAME applications",
```
Comment (robertgmoss) [inline commands/applications.go:26]: FYI there is a neat little trick going on here: `BROOKLYN_NAME` is automatically substituted with the binary's name in the help text, so recommend you keep that bit.

File: commands/effectors.go [hunk 2]
```diff
@@ -21,9 +21,9 @@ func NewEffectors(network *net.Network) (cmd *Effectors) {
 
 func (cmd *Effectors) Metadata() command_metadata.CommandMetadata {
 	return command_metadata.CommandMetadata{
-		Name:        "effectors",
+		Name:        "effector",
 		Description: "Show the list of effectors for an application and entity",
-		Usage:       "BROOKLYN_NAME effectors APPLICATION ENITITY",
```
Comment (robertgmoss) [inline commands/effectors.go:26]: same here

File: commands/entities.go [hunk 3]
```diff
@@ -20,9 +20,9 @@ func NewEntities(network *net.Network) (cmd *Entities) {
 
 func (cmd *Entities) Metadata() command_metadata.CommandMetadata {
 	return command_metadata.CommandMetadata{
-		Name:        "entities",
+		Name:        "entity",
 		Description: "Show the entites for an application",
-		Usage:       "BROOKLYN_NAME entities APPLICATION",
```
Comment (robertgmoss) [inline commands/entities.go:25]: and here

File: commands/sensors.go [hunk 4]
```diff
@@ -20,9 +20,9 @@ func NewSensors(network *net.Network) (cmd *Sensors) {
 
 func (cmd *Sensors) Metadata() command_metadata.CommandMetadata {
 	return command_metadata.CommandMetadata{
-		Name:        "sensors",
+		Name:        "sensor",
 		Description: "Show the sensors for an application and entity",
-		Usage:       "BROOKLYN_NAME sensors APPLICATION ENTITY",
```
Comment (robertgmoss) [inline commands/sensors.go:25]: and here

File: commands/applications.go (hunk 1 above)
Comment (robertgmoss) [inline commands/applications.go:26]: The substitution happens [here](https://github.com/brooklyncentral/brooklyn-cli/blob/cli-list/app/app.go#L31)

Comment (geomacy) [inline commands/applications.go:26]: In this branch the 'Application's class has moved to being a subcommand of 'list', so the BROOKLYN_NAME isn't actually right there.  The help text for list grabs the usage for each of its subcommands, so the help for list now looks like:

```
NAME:
   list - List details for a variety of operands

USAGE:
   bk list (application | entity | sensor | effector)
... list application            Show the status and location of running applications
... list entity APPLICATION     Show the entites for an application
... list sensor APPLICATION ENTITY  Show the sensors for an application and entity
... list effector APPLICATION ENTITY    Show the list of effectors for an application and entity
```

Comment (robertgmoss) [inline commands/applications.go:26]: Ok, I see what you are doing now.  Is it worth some kind of comment to say that this is a subcommand?

**Comment by robertgmoss** [comment]
LGTM merge when ready

File: commands/applications.go (hunk 1 above)
Comment (geomacy) [inline commands/applications.go:26]: Could well be, or even maybe worth renaming it as ApplicationSubCommand or some such.  However, given discussions yesterday we may well not be sticking with "verb noun" style, so this branch may be superseded.

Comment (robertgmoss) [inline commands/applications.go:26]: Maybe so, but adds good work related to subcommands, so still worth merging

Comment (geomacy) [inline commands/applications.go:26]: Ok - let me just do a few quick renames for clarity where something is a subcommand, and will push into this branch.

**Pull request merged**